### PR TITLE
Fix visibility container potentially getting in wrong visual state

### PR DESCRIPTION
**Pull request by peppy**
If a `VisibiltyContainer` was visible at the point of checking `ShowHidden` (`LoadCompleteAsync`), but then changed to `Hidden` in a parent's `LoadComplete` method, the `UpdateState` call would not be fired and the initial display state would be incorrect.

### CODE REVIEW

File: osu.Framework/Graphics/Containers/VisibilityContainer.cs [hunk 1]
```diff
@@ -31,12 +33,13 @@ protected override void LoadAsyncComplete()
                 // do this without triggering the StateChanged event, since hidden is a default.
                 PopOut();
                 FinishTransforms(true);
+                didInitialHide = true;
```
Comment (UselessToucan) [inline osu.Framework/Graphics/Containers/VisibilityContainer.cs]: Looks like a duplicate of a `Drawable.IsLoaded` property. Why did not you use it instead?

Comment (iiSaLMaN) [reply osu.Framework/Graphics/Containers/VisibilityContainer.cs]: You seem to be wrong here, `LoadAsyncComplete()` is invoked once [the drawable is currently loading](https://www.github.com/ppy/osu-framework/blob/master/osu.Framework/Graphics/Drawable.cs#L237-L266), while `LoadComplete()` on the other hand, it's invoked once the drawable has finished loading, in `Ready` state and an `UpdateSubTree()` has been called once. `Drawable.IsLoaded` property simply checks whether the drawable is in a `Loaded` state, and that is only set on [the load completion of the drawable](https://github.com/ppy/osu-framework/blob/4d58344fb52493cc7022c47ddbdec9fea5c04c98/osu.Framework/Graphics/Drawable.cs#L277-L291), I cannot think of a way to check if the container is in initial hidden state without setting a flag there.

Comment (UselessToucan) [reply osu.Framework/Graphics/Containers/VisibilityContainer.cs]: I know it. The idea is that the problem probably should be resolved on a `Drawable` class level.

Comment (peppy) [reply osu.Framework/Graphics/Containers/VisibilityContainer.cs]: how?

Comment (UselessToucan) [reply osu.Framework/Graphics/Containers/VisibilityContainer.cs]: I did not think about it in details. I guess the simple and straightforward solution is to introduce `Drawable.IsReady`(as far as I understand, `Drawable.loadState` is `Ready` after `LoadAsyncComplete` execution) property and use it in this case.

Comment (iiSaLMaN) [reply osu.Framework/Graphics/Containers/VisibilityContainer.cs]: Though come to think of it, checking `!StartHidden` should be enough for that, no?

Comment (peppy) [reply osu.Framework/Graphics/Containers/VisibilityContainer.cs]: no. because it's a property. which can change.

**Comment by UselessToucan** [comment]
> we need to rewrite this in the future

Do you know how this code should be rewritten?

**Comment by smoogipoo** [comment]
I don't. This container may potentially serve too large of an issue scope.

**Pull request merged**